### PR TITLE
Added Rabin_karp Algorithm in java language

### PR DESCRIPTION
**Pull request by Anshuvats**
The Rabin–Karp algorithm or Karp–Rabin algorithm is a string-searching algorithm that uses hashing to find any one of a set of pattern strings in a text. For text of length n and p patterns of combined length m, its average and best case running time is O(n+m) in space O(p), but its worst-case time is O(nm).

### CODE REVIEW

File: Rabin_carp_Algorithm/Rabin_carp_algorithm.java [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+    public static void main(String[] args) throws IOException
+    {    
+        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
+        System.out.println("Rabin Karp Algorithm Test\n");
+        System.out.println("\nEnter Text\n");
+        String text = br.readLine();
+        System.out.println("\nEnter Pattern\n");
+        String pattern = br.readLine();
+        System.out.println("\nResults : \n");
+        RabinKarp rk = new RabinKarp(text, pattern);
```
Comment (jainaman224) [inline Rabin_carp_Algorithm/Rabin_carp_algorithm.java]: What does this line do?
Can we replace it with only function call?

File: Rabin_carp_Algorithm/Rabin_carp_algorithm.java [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+public class RabinKarp 
+{
+    /** String Pattern **/
+    private String pat; 
+    /** pattern hash value **/    
+    private long patHash;    
+    /** pattern length **/
+    private int M;  
+    /** Large prime **/         
+    private long Q;
```
Comment (jainaman224) [inline Rabin_carp_Algorithm/Rabin_carp_algorithm.java]: Please use descriptive variable names.

**Comment by Anshuvats** [comment]
okay,sir. i will update asap

File: Rabin_carp_Algorithm/Rabin_carp_algorithm.java (hunk 1 above)
Comment (Anshuvats) [reply Rabin_carp_Algorithm/Rabin_carp_algorithm.java]: yes. why not?

**Comment by jainaman224** [comment]
@Anshuvats If you need some help you can ask on public channel.

**Comment by Anshuvats** [comment]
okk,sir

**Pull request closed without merging**